### DISCUSSION TITLE
Leptos for a cross platform app

### DISCUSSION THREAD

**Discussion by KelvinJPS** [leptos-rs/leptos#793] [General]
So I saw the example of letpos with GTK, rust currently has good support for accesing the os apis and ui toolkits, for example there is GTK and winrs https://github.com/microsoft/windows-rs, my question is can I use leptos as abtraction to updating  changes in the UI toolkits, for example GTK, windows, or even android ?

**Comment by KelvinJPS** [comment]
and would leptos increase the  size to the final  binary?

**Comment by gbj** [comment]
For sure! We don't have a renderer or anything for most platforms, but most other UI toolkits are built on the same model as the DOM: stateful, retained-mode widgets that fire events or callbacks. You can do things like the GTK example to use `leptos_reactive` to drive changes to any UI library like this, although you'll have to do a lot of work yourself. 

But it's very viable: Lapce, for example, is working on a [new UI library called `floem`](https://github.com/lapce/floem) driven by Leptos reactivity.

**Comment by KelvinJPS** [comment]
thanks,  I really like how simple and easy the reactive system  is
